DSpace/dspace-angular
# Adding/removing relationships in the submission form does not update the view automatically

## References
* Fixes #1671
* Requires DSpace/DSpace#8848

## Description
This PR fixes 2 problems The first problem is that the relationships list on submission forms is not updated when adding or removing new items. The second issue is that by default relationship fields can't retrieve their current scope in `SubmissionSectionFormComponent.inCurrentSubmissionScope()`.

## Instructions for Reviewers
**Changes for the submission form not updating the view when adding/removing relationships:**
* `SubmissionSectionFormComponent`: Use `sectionData` instead of `sectionMetadata` to retrieve the field value (because `sectionMetadata` did not include the relationships)
* Hide the repeatable field button for relationship fields instead of disabling it

**Change for the default relationship field not retrieving it's scope:**
In `SubmissionSectionFormComponent#inCurrentSubmissionScope` the scope was retrieved by checking if the `FormField` had a `selectableMetadata`, but relation fields don't have this field, they use the `selectableRelationship` field instead. The fix for this bug was just to add a custom if condition [here](https://github.com/DSpace/dspace-angular/blob/b318d7281ad9602f1afaf159f926391cc58f7ed6/src/app/submission/sections/form/section-form.component.ts#L262-L263) in order to retrieve the correct field from `formConfig`.

**Guidance for how to test this PR:**
**Testing the submission form not updating the view when adding/removing relationships:**
- Create a new item with `dspace.entity.type` `Person`
- Create a collection with `dspace.entity.type` `Publication`
- Add this field to the submission form of the `Publication` collection:
  ```xml
  <relation-field>
      <relationship-type>isAuthorOfPublication</relationship-type>
      <search-configuration>person</search-configuration>
      <repeatable>true</repeatable>
      <label>Author</label>
      <hint>Enter the author’s name (Family name, Given names).</hint>
      <required></required>
  </relation-field>
  ```
- Run `dspace initialize-entities -f dspace/config/entities/relationship-types.xml`
- Create a new item in that collection and check that you can see the author after adding it using the relation field without having to refresh the page
- Delete the author and check that it disappears again without needing to refresh the page

**Testing the default relationship field not retrieving it's scope:**
- This can't currently be tested yet through the UI because of another bug but you can test it using the debugger. When `SubmissionSectionFormComponent.inCurrentSubmissionScope()` is called for a relationship field, check that the scope that is retrieved is correct.

## Checklist
